internal/pkg/agent/cmd/enroll_cmd.go

```diff
@@ -328,26 +360,21 @@ func (c *enrollCmd) writeDelayEnroll(streams *cli.IOStreams) error {
 	return nil
 }
 
-func (c *enrollCmd) fleetServerBootstrap(ctx context.Context, persistentConfig map[string]interface{}) (string, error) {
-	c.log.Debug("verifying communication with running Elastic Agent daemon")
+func (c *enrollCmd) fleetServerBootstrap(ctx context.Context, persistentConfig map[string]interface{}, agentConErr error) (string, error) {
 	agentRunning := true
 	if c.options.FleetServer.InternalPort == 0 {
 		c.options.FleetServer.InternalPort = defaultFleetServerInternalPort
 	}
-	_, err := getDaemonState(ctx)
-	if err != nil {
-		if !c.options.FleetServer.SpawnAgent {
-			// wait longer to try and communicate with the Elastic Agent
-			err = wait.ForAgent(ctx, c.options.DaemonTimeout)
-			if err != nil {
-				return "", errors.New("failed to communicate with elastic-agent daemon; is elastic-agent running?")
-			}
-		} else {
-			agentRunning = false
+	if !c.options.FleetServer.SpawnAgent {
+		// wait longer to try and communicate with the Elastic Agent
+		if agentConErr != nil {
+			return "", errors.New("failed to communicate with elastic-agent daemon; is elastic-agent running?")
 		}
+	} else {
+		agentRunning = false
```

Comment:
i think this does not really keep the logic. 
we should check probably
```go
if agentConnErr != nil && !c.options.FleetServer.SpawnAgent {
    return "", errors.New("failed to communicate with elastic-agent daemon; is elastic-agent running?")
} else {
    agentRunning = agentConnErr == nil
}
```

Reply:
yes, you're right, updating right away